### ISSUE TITLE
AMA-aware autosens algorithm

### ISSUE THREAD

**Issue by scottleibrand** [openaps/oref0#98]
Now that AMA carb absorption calculations are being done using the same data used to detect sensitivity, it should be fairly straightforward to improve the autosens algorithm to directly exclude deviations caused by carb absorption (when carb info is avilable), thereby allowing us to use a much narrower percentile range and detect excess sensitivity or resistance with much more precision.

**Comment by scottleibrand** [comment]
It just occurred to me that we might be able to accomplish the same thing, without having to have carb info available, by excluding from the autosens calculation any deviations large enough to be likely caused by carb absorption, and then narrowing the percentile range.  For now I picked a deviation cutoff of 6 mg/dL per 5m, which quite accurately matches the deviations caused by carb absorption in my current data set, and narrowed the percentile range to p45-p50.

I've verified that this more accurately detects sensitivity, but will need to test it in resistance situations as well.
